### ISSUE TITLE
mnist-multi-gpu

### ISSUE THREAD

**Issue by xnejed07** [normanheckscher/mnist-multi-gpu#3]
Hello, I am trying to implement multi-GPU MNIST classifier from scratch. I have a problem with classifier poor accuracy which is about 50-60 percent. Do you have some recommendation how to improve my code ?
[multi_gpu.txt](https://github.com/normanheckscher/mnist-multi-gpu/files/1563650/multi_gpu.txt)

**Comment by normanheckscher** [comment]
Use the Keras API. I'm not sure if the make_parallel function has found its way into core tensorflow yet, although it's easy enough to copy the function call to your code manually from the Keras GitHub repo. I've moved my main workflow to Keras as it's the cleanest API. Keras has become part of core tensorflow as of 1.4

**Comment by normanheckscher** [comment]
Oh, and depending on your hardware, you won't see an increase in speed with mnist on multi GPU systems. It's so small that it's faster to load the entire system to ram rather than split the problem into a multi GPU problem.

**Comment by xnejed07** [comment]
Right now, I do not worry about processing speed. This is just education example for me, once it is working I'll use that for more complicated stuff. Problem is that optimization algorithm with averaging gradients from 2 gpu doesnt converge to standard accuracy for MNIST that should be about 99%. There must be some error with weight sharing or something like that, but I am new to TF so I do not know where to start.
